aws/cloudformation/iam.yml.erb

```diff
@@ -835,6 +840,309 @@ roles.each do |name, config| %>
+    Properties:
+      RoleName: github-actions-cloudformation-<%=environment_type%>-role
+      Description: Role for GitHub Actions to create/update CloudFormation stacks.
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Principal:
+              Federated: !Sub "arn:aws:iam::${AWS::AccountId}:oidc-provider/token.actions.githubusercontent.com"
+            Action: sts:AssumeRoleWithWebIdentity
+            Condition:
+              StringEquals:
+                # See: https://docs.github.com/en/actions/security-for-github-actions/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services#configuring-the-role-and-trust-policy
+                token.actions.githubusercontent.com:aud: sts.amazonaws.com
+                token.actions.githubusercontent.com:sub: 'repo:code-dot-org/code-dot-org:environment:marketing-sites-<%=environment_type%>'
+      Policies:
+        - PolicyName: github-actions-cloudformation-policy
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Sid: CloudFormationPermissions
+                Effect: Allow
+                Action:
+                  - cloudformation:CreateStack
+                  - cloudformation:DescribeStacks
+                  - cloudformation:CreateChangeSet
+                  - cloudformation:DescribeChangeSet
+                  - cloudformation:DeleteChangeSet
+                  - cloudformation:ExecuteChangeSet
+                  - cloudformation:ListStackResources
+                Resource: '*'
+              - Sid: PassRoleToCloudFormation
+                Effect: Allow
+                Action:
+                  - iam:PassRole
+                Resource: !GetAtt CloudFormationServiceMarketingSites<%=environment_type.capitalize%>Role.Arn
+
+  # Enforce least privilege on Roles created/passed/attached by the Role that provisions marketing site Stacks.
+  MarketingSitesRoleCreationBoundary<%=environment_type.capitalize%>Policy:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      Description: Permissions boundary for Roles created by CloudFormationMarketingSites<%=environment_type.capitalize%>Role
+      ManagedPolicyName: marketing-sites-role-permissions-boundary-<%=environment_type%>
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Sid: AllowSecretsManagerForMarketingSites
+            Effect: Allow
+            Action:
+              - secretsmanager:GetSecretValue
+              - secretsmanager:DescribeSecret
+            Resource: 'arn:aws:secretsmanager:*:*:secret:marketing-sites/<%=environment_type%>/*'
+          - Sid: AllowCloudfrontInvalidation
+            Effect: Allow
+            Action:
+              - cloudfront:CreateInvalidation
+            Resource: '*'
+            Condition:
+              StringEquals:
+                aws:ResourceTag/environment-type: <%=environment_type%>
```

Comment:
`cloudfront:CreateInvalidation` can be scoped down to specific distributions via `Resource`. Looks like you're using tagging instead?

Reply:
The Distribution ARN doesn't have any human readable identifiers. Just that randomly generated Distribution ID.

Reply:
ah, makes sense. I wonder if we should go even farther with standardized tagging?